app/uk/gov/hmrc/individualsemploymentsapi/connector/IfConnector.scala

```diff
@@ -92,41 +88,30 @@ class IfConnector @Inject()(servicesConfig: ServicesConfig, http: HttpClient, va
                          request: RequestHeader,
                          requestUrl: String)
                         (implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[A]] = x.recoverWith {
-    case validationError: JsValidationException => {
+    case validationError: JsValidationException =>
       logger.warn("Integration Framework JsValidationException encountered")
       auditHelper.auditIfApiFailure(correlationId, matchId, request, requestUrl, s"Error parsing IF response: ${validationError.errors}")
       Future.failed(new InternalServerException("Something went wrong."))
-    }
-    case Upstream4xxResponse(msg, 404, _, _) => {
+    case UpstreamErrorResponse(msg, 404, _, _) =>
       auditHelper.auditIfApiFailure(correlationId, matchId, request, requestUrl, msg)
 
       msg.contains("NO_DATA_FOUND") match {
         case true => Future.successful(Seq.empty)
-        case _    => {
+        case _    =>
           logger.warn("Integration Framework NotFoundException encountered")
           Future.failed(new NotFoundException(msg))
-        }
       }
-    }
-    case Upstream5xxResponse(msg, code, _, _) => {
-      logger.warn(s"Integration Framework Upstream5xxResponse encountered: $code")
-      auditHelper.auditIfApiFailure(correlationId, matchId, request, requestUrl, s"Internal Server error: $msg")
-      Future.failed(new InternalServerException("Something went wrong."))
-    }
```

Comment:
🚫 We display a different message/response on 5xx errors than 4xx.

Use `UpstreamErrorResponse.Upstream5xxResponse` as a replacement. `UpstreamErrorResponse` matches against all errors.

Reply:
OK, I'll go back and see about using that construct